src/libs/elektra/keyname.c

```diff
@@ -165,6 +165,52 @@
 #include "kdbhelper.h"
 #include "kdbinternal.h"
 
+/**
+ * Helper method: creates a deep copy of a keyname. Does not copy the refs counter.
+ * @param source the keyname to copy
+ * @return copied keyname
+ */
+struct _KeyName * keyNameCopy (struct _KeyName * source)
```

Comment:
```suggestion
static struct _KeyName * keyNameCopy (struct _KeyName * source)
```

Reply:
No, the `keyNameCopy` function is used outside `keyname.c` aswell.

Reply:
In that case, I'd move the `keyNameIncRef` into this function, because AFAICT, you always used `keyNameDecAndDel`, so having refcount zero would just be wrong.